### PR TITLE
POC'd interoperability using callable

### PR DESCRIPTION
**Pull request by MatthewBlanski**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

## Unpackaged Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: force-app/main/default/classes/HealthCheckVMapper.cls [hunk 1]
```diff
@@ -58,54 +58,114 @@ public virtual with sharing class HealthCheckVMapper {
...
+                );
+            } catch (InvalidParameterValueException ipve) {
+                //We expect this if the pattern is followed and something goes wrong
+                //We can swallow this at the Health Check Definition Model Level
+            } catch (Exception e) {
+                //This is for all the unexpected funsies
+                //We can swallow this at the Health Check Definition Model Level
+            }
+        }
+        return this.locateHealthCheckDefinitionService().getHealthCheckDefinitionModels(); // retrieve models
```
Comment (salvadorsfdc) [inline force-app/main/default/classes/HealthCheckVMapper.cls:143]: do we need to return healthCheckDefinitionModelListFromProduct  instead?

File: force-app/main/default/classes/HealthCheckVMapper.cls [hunk 2]
```diff
@@ -58,54 +58,114 @@ public virtual with sharing class HealthCheckVMapper {
...
+                String productHealthCheckDefinitionJSON = (String) ((Callable) classType.newInstance())
+                    .call(
+                        'Health Check Definition',
+                        new Map<String, Object>{ 'Api Verison' => productRegistryModel.apiVersion }
+                    );
+
+                List<HealthCheckDefinitionModel> healthCheckDefinitionModelListFromProduct = (List<HealthCheckDefinitionModel>) JSON.deserialize(
+                    productHealthCheckDefinitionJSON,
+                    List<HealthCheckDefinitionModel>.class
+                );
```
Comment (salvadorsfdc) [inline force-app/main/default/classes/HealthCheckVMapper.cls:134]: productRegistryModelList.addAll(healthCheckDefinitionModelListFromProduct);

and return productRegistryModelList;

Comment (MatthewBlanski) [reply force-app/main/default/classes/HealthCheckVMapper.cls:134]: That's accurate. Seems I missed that step in POCing it.

File: force-app/main/default/classes/HealthCheckVMapper.cls (hunk 1 above)
Comment (MatthewBlanski) [reply force-app/main/default/classes/HealthCheckVMapper.cls:143]: We do

**Pull request closed without merging**